src/main_dpdk.cpp

```diff
@@ -5951,6 +5978,24 @@ COLD_FUNC int update_global_info_from_platform_file(){
         }
     }
 
+    if (cg->m_latency_measurement.length()) {
+        if ((strcasecmp(cg->m_latency_measurement.substr(0, 5).c_str(), "NANOS") == 0) ||
+            (strcmp(cg->m_latency_measurement.c_str(), "1") == 0)) {
+            g_opts->m_latency_measurement = CParserOption::LATENCY_METHOD_NANOS;
+        }
+    }
+
+    if (cg->m_timesync_method.length()) {
+        if ((strcasecmp(cg->m_timesync_method.c_str(), "PTP") == 0) ||
+            (strcmp(cg->m_timesync_method.c_str(), "1") == 0)) {
+            g_opts->m_timesync_method = CParserOption::TIMESYNC_PTP;
```

Comment:
If we have std::string i think ther is no need to fallback to posix functions.

Maybe something like this?
```
        std::transform(cg->m_timesync_method.begin(), cg->m_timesync_method.end(), cg->m_timesync_method.begin(), ::toupper);

        if ((cg->m_timesync_method.compare("PTP") == 0) || 
            (cg->m_timesync_method.compare("1") == 0)){
            g_opts->m_timesync_method = CParserOption::TIMESYNC_PTP;
        }
```

Reply:
IMO Posix is much easier to understand.  Readibility counts

Reply:
For me `.compare()` is more natural way, but if you want, you can keep it as is.

Btw. I think the best should be doing toupper/tolower transofrmation while reading, parameters and in logic just compare it.